tests/cloudpickle_test.py

```diff
@@ -2130,6 +2137,38 @@ def test_pickle_importable_typevar(self):
         from typing import AnyStr
         assert AnyStr is pickle_depickle(AnyStr, protocol=self.protocol)
 
+    @unittest.skipIf(sys.version_info < (3, 7),
+                     "Pickling generics not supported below py37")
+    def test_generic_type(self):
+        for obj in _generic_objects_to_test():
+            pickle_depickle(obj, protocol=self.protocol)
+
+    @unittest.skipIf(sys.version_info < (3, 7),
+                     "Pickling type hints not supported below py37")
+    def test_locally_defined_class_with_type_hints(self):
```

Comment:
The name used in this test is a little bit confusing: are we testing classes inheriting from `Generic` constructs, or the pickling of `Generic` type hits? it is not the same, and I am not sure why we should test both things in the same tests. See my comments below for additional remarks.

Reply:
I my initial commit, this test was testing for all kinds of type annotations, both simple literal types and more complex custom types based on generics and their subclasses.